pycbc/filter/matchedfilter.py

```diff
@@ -1942,7 +1960,118 @@ def compute_followup_snr_series(data_reader, htilde, trig_time,
                            onsource_idx + half_dur_samples + 1)
     return snr[onsource_slice] * norm
 
-__all__ = ['match', 'matched_filter', 'sigmasq', 'sigma', 'get_cutoff_indices',
+def optimized_match(
+    vec1,
+    vec2,
+    psd=None,
+    low_frequency_cutoff=None,
+    high_frequency_cutoff=None,
+    v1_norm=None,
+    v2_norm=None,
+    return_phase=False,
+):
+    """Given two waveforms (as numpy arrays),
+    compute the optimized match between them, making use
+    of scipy.minimize_scalar.
+
+    This function computes the same quantities as "match";
+    it is more accurate and slower.
+
+    Parameters
+    ----------
+    vec1 : TimeSeries or FrequencySeries
+        The input vector containing a waveform.
+    vec2 : TimeSeries or FrequencySeries
+        The input vector containing a waveform.
+    psd : FrequencySeries
+        A power spectral density to weight the overlap.
+    low_frequency_cutoff : {None, float}, optional
+        The frequency to begin the match.
+    high_frequency_cutoff : {None, float}, optional
+        The frequency to stop the match.
+    v1_norm : {None, float}, optional
+        The normalization of the first waveform. This is equivalent to its
+        sigmasq value. If None, it is internally calculated.
+    v2_norm : {None, float}, optional
+        The normalization of the second waveform. This is equivalent to its
+        sigmasq value. If None, it is internally calculated.
+    return_phase : {False, bool}, optional
+        If True, also return the phase shift that gives the match.
+
+    Returns
+    -------
+    match: float
+    index: int
+        The number of samples to shift to get the match.
+    phi: float
+        Phase to rotate complex waveform to get the match, if desired.
+    """
+
+    from scipy import integrate
+    from scipy.optimize import minimize_scalar
+
+    htilde = make_frequency_series(vec1)
+    stilde = make_frequency_series(vec2)
+
+    # a first time shift to get in the nearby region;
+    # then the optimization is only used to move to the
+    # correct subsample-timeshift witin (-delta_t, delta_t)
+    # of this
+    _, max_id, _ = match(
+        htilde,
+        stilde,
+        psd=psd,
+        low_frequency_cutoff=low_frequency_cutoff,
+        high_frequency_cutoff=high_frequency_cutoff,
+        return_phase=True
+    )
+    stilde = stilde.cyclic_time_shift(-max_id * stilde.delta_t)
+
+    frequencies = stilde.sample_frequencies.numpy()
+    waveform_1 = htilde.numpy()
+    waveform_2 = stilde.numpy()
+    fmin = low_frequency_cutoff if low_frequency_cutoff is not None else -numpy.inf
+    fmax = high_frequency_cutoff if high_frequency_cutoff is not None else numpy.inf
+
+    mask = numpy.logical_and(fmin <= frequencies, frequencies < fmax)
```

Comment:
I think this mask needs to be applied to the frequencies array, otherwise I think the code will fail in the case that low_frequency_cutoff *or* high_frequency_cutoff is given.

Reply:
Thanks for noticing this issue! I have fixed it.